js/builder_graph.js

```diff
@@ -445,20 +445,36 @@ class PowderInputNode extends InputNode {
 
     compute_func(input_map) {
         // TODO: haha improve efficiency to O(n) dumb
-        // also, error handling is missing
         let input = this.input_field.value.trim();
         let powdering = [];
         let errorederrors = [];
         while (input) {
             let first = input.slice(0, 2);
             let powder = powderIDs.get(first);
             if (powder === undefined) {
-                return null;
+                if (first.length > 0) {
+                    errorederrors.push(first);
+                } else {
+                    break;
+                }
             } else {
                 powdering.push(powder);
             }
             input = input.slice(2);
         }
+
+        if (this.input_field.getAttribute("placeholder") != null) {
+            let placeholder = this.input_field.getAttribute("placeholder");
+            if (placeholder.includes(" slots") && parseInt(placeholder[0]) < powdering.length)
+                errorederrors.push("Too many powders: " + powdering.length);
+        }
+
+        if (errorederrors.length) {
```

Comment:
Is errorederrors being used anywhere else?

Reply:
Just in this function.
It was there before I added any error detection (which is why I used it) and I can certainly remove it - but if you plan on having error handling (displaying the exact error) in the future, it might be useful then....

Reply:
I see, TY

probably useful enough to keep